File: leetcode-problems-with-programming-solutions/python/array-101-problems-with-solutions/check_if_exist.py

"""
Given an array arr of integers, check if there exist two indices i and j such that :

    * i != j
    * 0 <= i, j < arr.length
    * arr[i] == 2 * arr[j]
 """


# Time Complexity: O(n^2)
class Solution(object):
    def __init__(self) -> None:
        pass
    
    def checkIfExist(arr):
            """
            :type arr: List[int]
            :rtype: bool
            """
            
            for i in range(len(arr)):
                for j in range(len(arr)):
                    #if i is not equal to j AND position i is equal to 2 * position j
                    if i != j and arr[i] == 2*arr[j]:
                        return True
                return False
           

#test with any array input          
arr = [3,1,7,5]

print(Solution.checkIfExist(arr))

#Output => False  |bcs there is no even number existing in the list|

File: leetcode-problems-with-programming-solutions/c++/array-101-problems-with-solutions/check_if_exist.cpp
#include <iostream>
using namespace std;

/*
Given an array arr of integers, check if there exist two indices i and j such that :

    * i != j
    * 0 <= i, j < arr.length
    * arr[i] == 2 * arr[j]
*/

bool checkIfExist(int arr[], int size){

    // int len = sizeof(arr);
    // cout << arr;

    for(int i = 0; i < size; i++){
        for (int j = 0; j < size; j++){
            //if i is not equal to j and position i in array is equal to 2 * position j in the array
            //then there is a even number in the array
            if(i != j && arr[i] == 2*arr[j]){
                return true;
            }
            
        }
       

    }
    return false;

};

int main()
{
    int arr[4] = {10, 5, 7, 9};
    int size = sizeof(arr) / sizeof(arr[0]);

    if(checkIfExist(arr, size)){
        cout << "True, i is exactly twice the value of j";
    }
    else{
        cout << "False, i is not exactly twice the value of j";
    }

    return 0;
}